File: ch2/src/main/java/org/study/actor/PongActor.java
package org.study.actor;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

/**
 * 임의의 문자열 혹은 "ping" 메시지를 받으면 "pong" 메시지를 보내는 퐁액터
 * @author Baekjun Lim
 */
public class PongActor extends UntypedActor {

    private LoggingAdapter log = Logging.getLogger(getContext().system(), this);
	private ActorRef ping; 
	
	public PongActor(ActorRef ping) {
		this.ping = ping;
	}
	
	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof String) {
	       	String msg = (String)message;
	       	log.info("Pong received {}", msg);
	       	ping.tell("pong", getSelf());
            Thread.sleep(1000); // 실전에서는 절대 금물!!!
		}
	}
	
}

File: scala-ch2/src/main/scala/org/study/actor/PongActor.scala
package org.study.actor

import akka.actor.Actor
import akka.actor.actorRef2Scala
import akka.event.Logging

class PongActor extends Actor {
  val log = Logging(context.system, this)
  def receive = {
    case msg @ "Ping" => { 
      log.info("Pong received {}", msg); 
      sender ! "Pong" 
    }
    case x @ _ => log.info(">>> received {}", x)
  }
}